src/Microsoft.Health.Fhir.SqlServer/Features/Search/QueryPlanCache/Ewma.cs

```diff
@@ -64,29 +64,32 @@ public void Update(string config, double newScore)
 
         public string GetBestMetric()
         {
-            lock (_lock)
-            {
-                string metricName;
+            string metricName;
 
-                // To make an informed decision, we need to have at least one score for each metric.
-                // With the current logic, we will execute each metric in a round-robin fashion until we reach the minimum number of iterations.
-                if (_iterations < _minIterationsForDecision)
-                {
-                    metricName = _metricNames[_iterations % _metricNames.Length];
-                }
-                else
+            if (_iterations < _minIterationsForDecision)
+            {
+                lock (_lock)
                 {
-                    metricName = _ewmaScores
-                        .Where(score => score.Value != null)
-                        .OrderBy(score => score.Value)
-                        .Select(score => score.Key)
-                        .FirstOrDefault();
+                    // To make an informed decision, we need to have at least one score for each metric.
+                    // With the current logic, we will execute each metric in a round-robin fashion until we reach the minimum number of iterations.
+                    if (_iterations < _minIterationsForDecision)
+                    {
+                        metricName = _metricNames[_iterations % _metricNames.Length];
+                        _iterations++;
+                        return metricName;
+                    }
                 }
+            }
 
-                _iterations++;
+            metricName = _ewmaScores
```

Comment:
What does this return if this is reached before any scores are set? For example if GetBestMetric is called enough times so that _interactions is greater than _minInteractionsForDecision, but Update hasn't been called yet.